Add support for the Ticwatch Pro(catfish)

MagneFire opened the pull request.

MagneFire committed:
Rename platform ray to catfish.

MagneFire committed:
linux-catfish: Switch to catfish oreo kernel.

MagneFire committed:
initrdscripts: Use correct partition for userdata.

MagneFire committed:
linux-catfish: Enable USB gadget.

MagneFire committed:
psplash: Select animated bootlogo relevant for screen size.
And remove local patch as it's now provided globally.

MagneFire committed:
initrdscripts: Don't restart psplash and explicitly start msm-fb-refresher.
msm-fb-refersher is now automatically executed before psplash if it exists on the ramdisk.

MagneFire committed:
linux-catfish: Disable secure IPC router.
Some android daemons use the IPC router but cannot use it unless it's set to insecure.

MagneFire committed:
linux-catfish: Backport Bluetooth 4.1 stack.

MagneFire committed:
linux-catfish: Enable Virtual HCI.

MagneFire committed:
linux-catfish: Fix CONFIG_BT_HCIUART_H4.
This appears to be needed for the Android daemon as that's the communication method between the SoC and the Bluetooth chip.

MagneFire committed:
linux-catfish: Fix tap-to-wake from deep sleep mode.

MagneFire committed:
base-files: Mount the correct /persist partition.

MagneFire committed:
android: Update system tar to support Bluetooth, WiFi and sensors.
This commit adds the needed files in order for Bluetooth, WiFi and sensors to get working.

The diff below shows the changed files and why they are changed/added.
diff '--color=auto' -ur system_back/etc/bluetooth/bt_did.conf system/etc/bluetooth/bt_did.conf: Import from catfish stock.
diff '--color=auto' -ur system_back/etc/bluetooth/bt_stack.conf system/etc/bluetooth/bt_stack.conf: Import from catfish stock.
diff '--color=auto' -ur system_back/etc/bluetooth/bt_vendor.conf system/etc/bluetooth/bt_vendor.conf: Added from catfish stock to ensure that the correct .bcd file is used.
Only in system/vendor/bin/hw: android.hardware.bluetooth@1.0-service: Import from catfish stock, needed by Bluebinder for Bluetooth support (Android daemon that interacts with hardware)
Only in system/vendor/bin: nanoapp_cmd: Import from catfish, needed for sensors (update firmware)
Only in system/vendor/bin: nanoapp_hr_log: Import from catfish, probably used for debugging
Only in system/vendor/bin: nanotool: Import from catfish, probably used for debugging
Only in system/vendor/etc/wifi: bcmdhd.cal: Import from catfish, contains configuration/calibration settings
Only in system/vendor/firmware: BCM43430B0.hcd: Bluetooth chip firmware
Only in system/vendor/firmware: bt_nv.bin: Contains MAC address for the Bluetooth chip
Only in system/vendor/firmware: fw_bcmdhd_apsta.bin: Import from catfish, probably also used for WiFi, though not found in logs.
Only in system/vendor/firmware: fw_bcmdhd.bin: Import from catfish, firmware for the WiFi chip.
Only in system/vendor/firmware: it7259.cfg: Import from catfish, probably used for touchscreen, though without it works also fine.
Only in system/vendor/firmware: it7259.fw: Import from catfish, probably used for touchscreen, though without it works also fine.
Only in system/vendor/firmware: nanohub.full.catshark.19061201.bin: Firmware for the nanohub sensor microcontroller.
Only in system/vendor/lib/hw: 1sensors.sawfish.so: Debug, backup from sawfish.
Binary files system_back/vendor/lib/hw/android.hardware.bluetooth@1.0-impl.so and system/vendor/lib/hw/android.hardware.bluetooth@1.0-impl.so differ
Only in system/vendor/lib/hw: sensors.nanohub.so: Identical to below, copied as the sensors.nanohub.so refers to this too.
Binary files system_back/vendor/lib/hw/sensors.sawfish.so and system/vendor/lib/hw/sensors.sawfish.so differ: Imported as sensors.nanohub.so from catfish.
Binary files system_back/vendor/lib/libbt-vendor.so and system/vendor/lib/libbt-vendor.so differ
Only in system/vendor/lib: libhubconnection.so: Import from catfish, needed by the sensors.nanohub.so.

MagneFire committed:
android-init: Start Bluetooth daemon and update sensor firmware.

MagneFire committed:
conf: Install Bluebinder to support Bluetooth.

MagneFire committed:
conf: Install heart rate monitor.

MagneFire committed:
conf: Install compass app.

MagneFire committed:
Remove wifi enabler service.
catfish uses a different kernel driver for WiFi as such we don't actually need this service.

jrtberlin commented on README-system-dir.md:
```diff
@@ -21,15 +21,15 @@ It is very likely that you are currently running an up-to-date version of WearOS
 
 Below you can find some links where you can find the system files for ME69F and update file for M9E41V:
-* M6E69F full system: https://androidfilehost.com/?fid=745425885120748838 or http://www.rootjunkysdl.com/files/Huawei%20Watch/Firmware/ray-m6e69f-wear-stock.zip
+* M6E69F full system: https://androidfilehost.com/?fid=745425885120748838 or http://www.rootjunkysdl.com/files/Huawei%20Watch/Firmware/catfish-m6e69f-wear-stock.zip
```
This is specific to sturgeon and should be either replaced or removed.

jrtberlin commented on README-system-dir.md:
````diff
@@ -21,15 +21,15 @@ It is very likely that you are currently running an up-to-date version of WearOS
 
 ```sh
-fastboot flash bootloader bootloader-ray-m6e69f.img
+fastboot flash bootloader bootloader-catfish-m6e69f.img
````
The "m6e69f" is probably sturgeon specific.

jrtberlin reviewed (commented):
Only a minor documentation error. Everything else looks good to merge.
Thanks!

MagneFire committed:
README-system-dir: Remove until we have instructions that are in line with the current method.

MagneFire committed:
android-init: Enable the LCD when screen is in the off state.
This line enables the LCD when the screen is in the off state, it is more energy effecient compared to the Always-on-Display.

MagneFire commented:
Thanks for the feedback!

I've removed the README until we have more accurate instructions.
Also added a patch so that the LCD is turned on whenever the screen transitions into the off state :smile:
